bot/cogs/snakes.py

```diff
@@ -40,8 +65,84 @@ def __init__(self, bot: AutoShardedBot):
+                         '&prop=extracts&exlimit=1&explaintext&format=json&formatversion=2'
+
+        async with ClientSession() as session:
+            async with session.get(possible_names) as all_sneks:
+                resp = str(await all_sneks.read(), 'utf-8')
+
+        # can we find a better way to do this? Doesn't seem too reliable, even though MW won't change their api. >(PC)
+        snek_list = resp[409:].lower().split('\\n')
+
+        # if name is None, choose a random snake. Need to clean up snek_list. >(PC)
+        if name is None:
+            name = choice(snek_list)
+
+        # stops the command if the snek is not on the list >(PC)
+        elif name.lower() not in snek_list:
+            await ctx.send('This is not a valid snake. Please request one that exists.\n'
+                           'You can find a list of existing snakes here: ')
+            return
+
+        # accounting for the spaces in the names of some snakes. Allows for parsing of spaced names. >(PC)
+        if name.split(' '):
+            name = '%20'.join(name.split(' '))
+
+        # leaving off here for the evening. Building the embed is easy. Pulling the information is hard. /s >(PC)
+        # snek_dict = await self.get_snek(name)
 
     # Any additional commands can be placed here. Be creative, but keep it to a reasonable amount!
+    @command()
+    async def play(self, ctx: Context, order):
+        """
+        DiscordPlaysSnek
+
+        Move the snek around the field and collect food.
+
+        Valid use: `bot.play {direction}`
+
+        With 'left', 'right', 'up' and 'down' as valid directions.
+        """
+
+        # Maybe one game at a given time, and maybe editing the original message instead of constantly posting
+        # new ones? Maybe we could also ask nicely for DMs to be allowed for this if they aren't. >(PC)
+        if order in self.movement_command.keys():
+            self.movement_command[order] += 1
+
+            # check if it's time to move the snek
+            if time() - self.last_movement > self.wait_time:
+                direction = max(self.movement_command,
+                                key=self.movement_command.get)
+                percentage = 100*self.movement_command[direction]/sum(self.movement_command.values())
+                move_status = self.game.move(direction)
+
+                # end game
+                if move_status == "lost":
+                    await ctx.send("We made the snek cry! :snake: :cry:")
+
+                # prepare snek message
+                snekembed = Embed(color=Color.red())
+                snekembed.add_field(name="Score", value=self.game.score)
+                snekembed.add_field(name="Winner movement",
+                                    value="{dir}: {per:.2f}%".format(dir=direction, per=percentage))
+
+                snek_head = next(emoji for emoji in ctx.guild.emojis if emoji.name == 'python')
```

Comment:
Very clever usage of next, I like that. However this creates issues if the name is changed. Might want to make this optional.